fathom/src/surface/elaboration/unification.rs

```diff
@@ -301,17 +311,19 @@ impl<'arena, 'env> Context<'arena, 'env> {
         &mut self,
         spine0: &[Elim<'arena>],
         spine1: &[Elim<'arena>],
-    ) -> Result<(), Error> {
+    ) -> Result<(), Error<'arena>> {
         if spine0.len() != spine1.len() {
-            return Err(Error::Mismatch);
+            return Err(Error::SpineMismatch);
         }
```

Comment:
This could be moved after the for loop if we wanted to pick up on errors in the spine more eagerly.